Source/Urho3D/Actions/ActionBuilder.h

```diff
@@ -0,0 +1,195 @@
+//
+// Copyright (c) 2021 the rbfx project.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+#include "AttributeAction.h"
+#include "CallFunc.h"
+
+namespace Urho3D
+{
+namespace Actions
+{
+class ActionCallHandler;
+}
+
+class ActionManager;
+
+/// Action as resource
+class URHO3D_API ActionBuilder
+{
+public:
+    /// Construct.
+    explicit ActionBuilder(Context* context);
+
+    /// Continue with provided action.
+    ActionBuilder& Then(const SharedPtr<Actions::FiniteTimeAction>& nextAction);
+
+    /// Run action in parallel to current one.
+    ActionBuilder& Also(const SharedPtr<Actions::FiniteTimeAction>& parallelAction);
+
+    /// Continue with MoveBy action.
+    ActionBuilder& MoveBy(float duration, const Vector3& offset, const ea::string& attributeName = Actions::POSITION_ATTRIBUTE);
```

Comment:
For new API we aim to use `ea::string_view`. This gives us a cheap way to avoid extra copy of a string when we pass raw C strings.